src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -216,6 +217,7 @@ void WorkStealingThreadPool::WorkStealingThreadPoolImpl::PrepareFork() {
   thread_count()->BlockUntilThreadCount(CounterType::kLivingThreadCount, 0,
                                         "forking", &work_signal_);
   lifeguard_.BlockUntilShutdown();
+  lifeguard_.Reset();
```

Comment:
Let's merge Reset and BlockUntilShutdown? - given it's not safe to call Reset until immediately after BlockUntilShutdown

Reply:
Done